enhancements/sig-architecture/81-addon-lifecycle/README.md

```diff
@@ -0,0 +1,669 @@
+# Add-on Lifecycle
+
+## Release Signoff Checklist
+
+- [] Enhancement is `implemented`
+- [] Design details are appropriately documented from clear requirements
+- [] Test plan is defined
+- [] Graduation criteria for dev preview, tech preview, GA
+- [] User-facing documentation is created in [website](https://github.com/open-cluster-management-io/open-cluster-management-io.github.io/)
+
+## Summary
+
+This proposal is to propose an enhancement of the `ClusterManagementAddon` and `ManagedClusterAddOn` API so that users can do
+canary upgrade on a specific addon.
+
+## Motivation
+
+Today, the addon agent is controlled by the addon manager, and when the addon manager upgrades all the agents will upgrade automatically.
+In some cases, we are seeing requirements that the user would like to rolling upgrade an addon to the new version. That is, 
+upgrade the addon on a group of clusters first, if everything works fine, then upgrade all the clusters to the new version.
+
+### Goals
+
+- Enhance the existing `ClusterManagementAddon` and `ManagedClusterAddOn` API, to support version-based addon installation 
+on a group of clusters. 
+- Enhance the existing `ClusterManagementAddon` and `ManagedClusterAddOn` API, to reflect the addon installation, upgrade 
+and rollback progress, including the addon version, progress, and completed time.
+
+### Non-Goals
+How to enhance the existing addon framework and addon manager to support multiple version addon installtion is not this 
```

Comment:
At a practical level in openshift, we found a need to tolerate multiple versions of various components at the same time.  Given a long rollout, potential canary failure, and then a roll-forward or roll-back, I suspect you'll face similar.